mlir/utils/jenkins/Dockerfile

```diff
@@ -100,6 +100,12 @@ RUN groupadd -g 109 render
 # Add ROCm build distribution
 RUN wget --no-check-certificate -qO - http://repo.radeon.com/rocm/rocm.gpg.key 2>/dev/null | apt-key add -
 RUN echo "deb [arch=amd64] $ROCM_DEB_REPO $ROCM_BUILD_NAME $ROCM_BUILD_NUM" > /etc/apt/sources.list.d/rocm.list
+RUN echo 'Package: *' > /etc/apt/preferences.d/repo-radeon-pin-600 && \
+    echo 'Pin: release o=repo.radeon.com' >> /etc/apt/preferences.d/repo-radeon-pin-600 && \
+    echo 'Pin-Priority: 600' >> /etc/apt/preferences.d/repo-radeon-pin-600
+RUN add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && \
+    apt-get install -y libpython3.8
```

Comment:
Added ppa:deadsnakes to install libpython3.8, as Ubuntu 22.04 comes with Python 3.10 by default, but rocgdb requires version 3.8.

Reply:
No it doesn't?
```
 % apt show rocm-gdb
Package: rocm-gdb
Version: 14.1.60102-119~22.04
Priority: optional
Essential: no
Section: utils
Maintainer: ROCm Debugger Support <rocm-gdb.support@amd.com>
Installed-Size: unknown
Depends: libexpat1, libtinfo5, libncurses5, rocm-dbgapi, libpython3.10 | libpython3.8, libbabeltrace-ctf1 (>= 1.2.1), libbabeltrace1 (>= 1.2.1), rocm-core
```

See that last line

Reply:
"docker build" disagrees:

7.291 The following packages have unmet dependencies:
7.463  rocm-gdb : Depends: libpython3.8 (>= 3.8.2) but it is not installable

Installing libpython3.10 doesn't solve it.

Reply:
... did you accidentally install the 20.04 version of the ROCm repository?

Reply:
Grr.  I think you're right.  Jammy vs Focal.  Silly Ubuntu names.  Still needed to install libpython3.10, but didn't need the ppa or 3.8.

Reply:
So we probably don't need the pin above either

Reply:
We still do, because solves a different problem, the version skew between rocm tools in the rocm repo and rocm tools in the ubuntu repo, for rocm-cmake and rocm-device-libs.  That's pretty much AMD's fault -- for some reason we have rocm-cmake6.2.0 as rocm-cmake6.2.0_0.13.0.60200-66~22.04_amd64.deb but rocm-cmake as rocm-cmake_0.13.0.60200-66~22.04_amd64.deb.  Ubuntu universe has rocm-cmake at 5.0.0 from rocm 5 and the 0.13.0 confuses apt.

Reply:
... we might want to just use `amdgpu-install` here?

Reply:
... Ah, yeah, that creates the pin, this makes sense

Reply:
I'll approve after we've got the deadsnakes invocation out of here